### PR TITLE
tests(pricing-table): add e2e storybook tests

### LINKED ISSUE #8780: [Test Scenario]: Pricing Table (Default, Without row headers, With subheaders)
### Component and test scenario title

Pricing Table - (Default, Without row headers, With subheaders)

### Test scenario steps

**Engineering info:**  
- Innovation Team engineer: 
- Innovation Team JIRA Ticket: 
- DPO consulting engineer: 

-----

### Component and test scenario title

Pricing Table - (Default, Without row headers, With subheaders)
https://carbon-design-system.github.io/carbon-for-ibm-dotcom/canary/web-components-experimental/?path=/story/components-pricing-table--default

### Test scenario steps

**Pricing Table - Default**

1. Ensure that the CTA buttons are clickable
2. Ensure that the table annotation toggle is working as expected.
3. Ensure that the header cell within each row is rendering.
4. Ensure that the highlighted column is set
5. Ensure that the horizontal scroll is working as expected when viewing on small breakpoints (eg. mobile, tablet).

**Pricing Table - Without row headers**

1. Same as default

**Pricing Table - Without subheaders**

1. Same as default
2. Check and make sure the subheaders are rendering.

### PR DESCRIPTION
**Pull request by jkaeser**
### Related Ticket(s)

Resolves #8780

### Description

Adds e2e test coverage for the `Pricing table` component in the `web-components` package.

### Changelog

**New**

- Adds e2e tests for the `dds-pricing-table` web component

**Changed**

- Fixes stable selectors for various `dds-pricing-table` components.
- Prevents calls to an undefined method when checking for empty cells.

### CODE REVIEW

File: packages/web-components/src/components/pricing-table/pricing-table-cell-annotation.ts [hunk 1]
```diff
@@ -9,12 +9,13 @@
...
 import { customElement, LitElement, property, html } from 'lit-element';
 import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings';
+import StableSelectorMixin from '../../globals/mixins/stable-selector';
 import styles from './pricing-table.scss';
 
 const { stablePrefix: ddsPrefix } = ddsSettings;
 
 @customElement(`${ddsPrefix}-pricing-table-cell-annotation`)
-class DDSPricingTableCellAnnotation extends LitElement {
+class DDSPricingTableCellAnnotation extends StableSelectorMixin(LitElement) {
```
Comment (proeung) [inline packages/web-components/src/components/pricing-table/pricing-table-cell-annotation.ts:18]: There will be a merge conflict with the changes in this PR (https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/8993/files#diff-2b05748087218281bf3f37742976a0308c629db96f8ac0e9a745bc60483cc55cR18).

**Review by proeung — COMMENTED** [review]
There are quite a few selector changes in this PR that might conflict with (https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/8993), however, we can resolve that depending on which PR gets merged down first.

Also, the e2e testing operation was canceled (See attached). I'm going to update this PR with `main` and see if the test goes through successfully.

Screen Shot 2022-06-16 at 12 48 29 PM

**Comment by proeung** [comment]
@jkaeser The test is still failing. Can you take a look? https://github.com/carbon-design-system/carbon-for-ibm-dotcom/runs/6922756481?check_suite_focus=true#step:8:2779

**Comment by jkaeser** [comment]
> @jkaeser The test is still failing. Can you take a look? https://github.com/carbon-design-system/carbon-for-ibm-dotcom/runs/6922756481?check_suite_focus=true#step:8:2779

@proeung @annawen1 Is it possible the environment these tests are running against does not include components behind feature flags? It looks to me like this stalls out when trying to `cy.get()` after `cy.visit()`.

**Comment by kennylam** [comment]
> @kennylam The e2e tests WC still time out despite re-running the job on GH Actions. Could this happen because this component is marked as a feature flag? Can you help take a look for us https://github.com/carbon-design-system/carbon-for-ibm-dotcom/runs/6991114023?check_suite_focus=true

Looks like the job is timing out because there's a 120 minute limit and the `pricing-table` tests never finish. If this is a blocker I suggest disabling these tests for now since it's behind a feature flag anyway.

**Comment by jkaeser** [comment]
@kennylam Just after posting that question last, I saw Andy put together this commit to do exactly what we need – https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/9064/commits/fd8e76def88a378b953c540407d0e959cdafe988. Going to give this a try here.

**Comment by proeung** [comment]
@jkaeser We're close to getting the e2e tests CI to run successfully! Looks like a test for the scroll horizontally behavior is failing. Can you take a look? https://github.com/carbon-design-system/carbon-for-ibm-dotcom/runs/7306500886?check_suite_focus=true#step:8:2793

Screen Shot 2022-07-13 at 1 08 33 PM

**Comment by jkaeser** [comment]
Added the "ready to merge" label back to this after fixing conflicts and integrating @andy-blum's fixed header tests. Just needs someone from @carbon-design-system/carbon-for-ibm-com-members's approval and we can finally get this thing merged 😄

**Pull request merged**